lgc/builder/ImageBuilder.cpp

```diff
@@ -1738,6 +1741,27 @@ void ImageBuilder::combineCubeArrayFaceAndSlice(Value *coord, SmallVectorImpl<Va
   coords.pop_back();
 }
 
+// =====================================================================================================================
+// A buffer descriptor may be incorrectly given when it should be an image descriptor, we need to fix it to valid buffer
+// type (0) to make hardware happily ignore it.
```

Comment:
Please mention that this is only needed for gfx10, and that justifies using hard-coded bit offsets here.

Reply:
sure, will add more detailed comment as suggested. What does 'PSO' stands for?

> The commit message, and somewhere in the code, should say that this is a workaround for a buggy application that is breaking the Vulkan spec by putting a buffer into the PSO then using it as an image in the shader.

Reply:
PSO stands for pipeline state option.